uitools/import/Esri/ArcGISRuntime/Toolkit/FloorFilter.qml

```diff
@@ -0,0 +1,254 @@
+
+/*******************************************************************************
+ *  Copyright 2012-2022 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+import Esri.ArcGISRuntime.Toolkit.Controller 100.14
+import QtQuick 2.12
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+
+
+/*!
+  \qmltype FloorFilter
+  \inqmlmodule Esri.ArcGISRuntime.Toolkit
+  \since 100.14
+  \brief Allows to display and filter the available floor aware layers in the current \c GeoView.
+*/
+Item {
+    id: floorFilter
+    property var geoView
+
+    property FloorFilterController controller: FloorFilterController {}
+
+    Binding {
+        target: controller
+        property: "geoView"
+        value: floorFilter.geoView
+    }
+
+    GridLayout {
+        columns: 2
+        ColumnLayout {
+            Layout.alignment: Qt.AlignBottom
+            ToolBar {
+                id: levelFilterMenu
+                visible: !internal.isFloorFilterCollapsed
+                // property var itemSelected: {text: ""}
+                property string itemSelected: ""
+
+                Action {
+                    id: close
+                    icon.source: "images/x.svg"
+                    onTriggered: {
+                        internal.isFloorFilterCollapsed = true
+                    }
+                }
+
+                ColumnLayout {
+
+                    ToolButton {
+                        Layout.fillWidth: true
+                        action: close
+                    }
+
+                    ToolSeparator {
+                        Layout.fillWidth: true
+                        orientation: Qt.Horizontal
+                    }
+
+                    Repeater {
+                        id: repeater
+                        property int downItem
+                        model: controller.levels
+                        delegate: ToolButton {
+                            Layout.fillWidth: true
+                            text: model.shortName
+                            onClicked: {
+                                repeater.setItemHighlighted(
+                                            index) // _q can access Repeater without using the id?
```

Comment:
Consider making use of the `AbstractButton.autoExclusive` to stop multiple buttons from being pressed at once.

```qml
ToolButton {
    autoExclusive: true
    onClicked: { downItem = this; checked = true; }
    highlighted: checked // May not be required?
```

Theoretically this will remove the need for `setItemHighlighted`.